velox/experimental/wave/common/tests/BlockTest.cu

```diff
@@ -11,7 +11,7 @@ __global__ void boolToIndices(
     int32_t** indices,
     int32_t* sizes,
     int64_t* times) {
-  extern __shared__ __align__(alignof(ScanAlgorithm::TempStorage)) char smem[];
+  __shared__ __align__(alignof(ScanAlgorithm::TempStorage)) char smem[1];
```

Comment:
Why do we need to change this?  It is required in CUDA: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#shared

Reply:
`extern` is a requirement for dynamic shared memory which is the case here.  I don't think the code still works because there is only 1 byte shared memory after the change.

Reply:
It doesn't need to be dynamic shared memory as the size is known at compile time. The use of `extern` requires to use different symbol names for the various `smem`. With these two factors, I am changing it to static shared memory.

Reply:
This is a legitimate use case of dynamic shared memory and the test is exploratory to make sure certain functionality is supported, so we want to keep the dynamic shared memory.  I did not get any error with `nvcc` 11.5, is it a new requirement from CUDA 12?  I would imagine the compiler should do some name mangling in this case.

Reply:
Ok, I'll revert that change. Let me compile again to document the error I was running into.

Reply:
```
[1789/1907] Building CUDA object velox/experim...iles/velox_wave_common_test.dir/BlockTest.cu.o
/workspace/velox-fb/velox/experimental/wave/common/tests/BlockTest.cu(44): warning #1886-D: specified alignment (8) is different from alignment (16) specified on a previous declaration
    extern __attribute__((shared)) __attribute__((aligned(alignof(cub::BlockReduce<int64_t, 256>::TempStorage))))
                                   ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"

/workspace/velox-fb/velox/experimental/wave/common/tests/BlockTest.cu(44): warning #1886-D: specified alignment (8) is different from alignment (16) specified on a previous declaration
    extern __attribute__((shared)) __attribute__((aligned(alignof(cub::BlockReduce<int64_t, 256>::TempStorage))))
                                   ^

Remark: The warnings can be suppressed with "-diag-suppress <warning-number>"
```

That warning indicates that given both variables are `extern` and named `smem`, the compiler is merging both of them, thus complaining they have different attributes.

Reply:
We can remove the alignment requirement here to get rid of the warnings.  These memory are always aligned to at least 256 bytes anyway.

Reply:
I've removed the alignment requirement and reverted to using dynamic shared memory.